Dockerfile

```diff
@@ -281,4 +281,37 @@ RUN set -eux \
 # clean 
     && cd / \
     && rm -rf /tmp/pg_repack-${PG_REPACK_VERSION} /tmp/pg_repack.zip \
-    && apk del .pg_repack-build-deps 
\ No newline at end of file
+    && apk del .pg_repack-build-deps 
+
+# Adding pgautofailover
+ARG PG_AUTO_FAILOVER_VERSION
+RUN set -eux \
+    && apk add --no-cache --virtual .pg_auto_failover-build-deps \
+        make \ 
+        gcc \
+        musl-dev \
+        krb5-dev \ 
+        openssl-dev \
+        clang15 \ 
+        ncurses-dev \
+        linux-headers \
+        zstd-dev \
+        lz4-dev \
+        zlib-dev \
+        libedit-dev \
+        libxml2-utils \
+        libxslt-dev \
+        llvm15 \
+# build pg_auto_failover
+    && wget  -O /tmp/pg_auto_failover-${PG_AUTO_FAILOVER_VERSION}.zip "https://github.com/hapostgres/pg_auto_failover/archive/refs/tags/v${PG_AUTO_FAILOVER_VERSION}.zip" \
+    && unzip  /tmp/pg_auto_failover-${PG_AUTO_FAILOVER_VERSION}.zip -d /tmp \
+    && ls -alh /tmp \
+    && cd /tmp/pg_auto_failover-${PG_AUTO_FAILOVER_VERSION} \
+    && make \
+    && make install \
+    && sed -i "s/\(shared_preload_libraries = '[^']*\)/\1,pgautofailover/"  /usr/local/share/postgresql/postgresql.conf.sample \
```

Comment:
We are already adding citus,postgis,pg_vector etc to shared_preload_libraries, can you add pgautofailover over there itself